File: dependency-inversion-typescript/src/fileUtilNotInverted.ts
import * as fs from "fs";

export function concatenateFiles(source1: string, source2: string, destination: string): void {
  const bytes1 = fs.readFileSync(source1);
  const bytes2 = fs.readFileSync(source2);
  const concatenated = Buffer.concat([bytes1, bytes2]);
  fs.writeFileSync(destination, concatenated);
}

File: dependency-inversion-python/src/file_util_not_inverted.py
from pathlib import Path

def concatenate_files(source1: Path, source2: Path, destination: Path) -> None:
    bytes1 = source1.read_bytes()
    bytes2 = source2.read_bytes()
    concatenate_files = bytes1 + bytes2
    destination.write_bytes(concatenate_files)

if __name__ == "__main__":
    concatenate_files(Path("a.txt"), Path("b.txt"), Path("out.txt"))1